backend/app/model/space_calculator.rb

```diff
@@ -0,0 +1,337 @@
+        lp_dims = Dimensions.new(location_profile.width,
+                                 location_profile.height,
+                                 location_profile.depth,
+                                 location_profile.dimension_units)
+      rescue MissingDimensionException
+        # the location's profile doesn't have the necessary data
+        @uncalculatable_locations << {'ref' => loc.uri, 'reason' => :location_lacks_dimension_data}
+        next
+      end
+
+      if cp_dims.bigger_in_any_than?(lp_dims)
+        # the container is bigger than the location in at least one dimension
+        @locations_without_space << {'ref' => loc.uri, 'reason' => :container_is_bigger}
+        next
+      end
+
+      # if we're still here then we'll have to run the packer to find out if it fits
+
+      # get a location packer
+      packer = LocationPacker.new(lp_dims)
+
+      # find the current containers at this location
+      containers = TopContainer.find_relationship(:top_container_housed_at).who_participates_with(loc)
+
+      # pack the current containers into the location
+      hit_error = false
+      containers.each do |container|
+        tcp = container.related_records(:top_container_profile)
+        if tcp.nil?
+          # need to say uncalculatable and get out
+          @uncalculatable_locations << {'ref' => loc.uri, 'reason' => :container_at_location_lacks_profile}
+
+          hit_error = true
+          break
+        end
+        unless packer.add_container(tcp.name,
+                                    Dimensions.new(tcp.width,
+                                                   tcp.height,
+                                                   tcp.depth,
+                                                   tcp.dimension_units),
+                                    tcp.stacking_limit)
+          # this shouldn't happen - we're just packing the containers already at the location
+          # we'll have to add it to uncalculatable and get out
+          @uncalculatable_locations << {'ref' => loc.uri, 'reason' => :containers_at_location_dont_fit}
+          hit_error = true
+          break
+        end
+      end
+
+      # If we missed any of the required data for this location, skip over it.
+      next if hit_error
+
+      # Now that the packer is pre-populated with the existing containers, we're
+      # ready to start trying to add our containers and count how many fit
+      count = 0
+      while packer.add_container(@container_profile.name,
+                                 Dimensions.new(@container_profile.width,
+                                                @container_profile.height,
+                                                @container_profile.depth,
+                                                @container_profile.dimension_units),
+                                 @container_profile.stacking_limit)
+        count += 1
+
+        # peace of mind
+        break if count > MAX_CONTAINERS_TO_PACK
```

Comment:
I am a little suspicious of this strategy, but this should at least be configurable so people can set their own limits for this.